### PR TITLE
 #169113440 edit or delete an accommodation facility

### PR DESCRIPTION
**Pull request by alexismajya**
#### What does this PR do?

- creates an `accommodation delete` endpoint 
- creates an`accommodation edit ` endpoint 
- creates a swagger documentation

#### Description of Task to be completed?

1) Have the following endpoints working:
-  `api/v1/accommodations/editAccommodation/<id>`
Optionaly, only a supplier can edit accommodation data from database including image/images
-  `api/v1/accommodations/deleteAccommodation/<id>`
The supplier can delete an accommodation data
2) Have a `swagger documentation` for those endpoints

#### How should this be manually tested?

After cloning this repository, run `npm run dev`. Then:
- Using `postman` test every endpoint above

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?

[Pivotal tracker story](https://www.pivotaltracker.com/story/show/169113440)

#### Screenshots (if appropriate)

Screen Shot 2019-10-21 at 16 17 36

#### Questions:

### CODE REVIEW

**Review by swaibat — COMMENTED** [review]
@alexismajya  great job but do mind first fixing hound and coveralls and codeclimate before reviews

**Review by lemoissonM — CHANGES_REQUESTED** [review]
please resolve the conflicts

**Review by swaibat — APPROVED** [review]

**Review by urbanishimwe — APPROVED** [review]

**Review by lemoissonM — APPROVED** [review]

File: src/middlewares/accommodationMiddleware.js [hunk 1]
```diff
@@ -1,6 +1,43 @@
...
+  req.imageArray = imageArray;
+  return next();
+};
 
 const Accommodation = {
+  checkUserSupplier(req, res, next) {
+    const { userData } = req;
+    if (userData.role === 'host') return next();
+    return sendResult(res, 401, 'You are not authorized');
+  },
```
Comment (elchroy) [inline src/middlewares/accommodationMiddleware.js]: This is not needed. We already have a middleware for this in `rolesMiddleware`

File: src/routes/accommodationRoute.js [hunk 2]
```diff
@@ -80,10 +80,16 @@ const fUpload = fileUpload({
   tempFileDir: path.join(__dirname, '../temp'),
 });
 
-const { checkForImages } = accMidd;
+const {
+  checkUserSupplier, isSupplierAccommodation, checkForImages, checkForImagesUpdate
```
Comment (elchroy) [inline src/routes/accommodationRoute.js]: Replare `checkUserSupplier ` with `checkHost` middleware from `rolesMiddleware`

File: src/tests/testAccomod.spec.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  it('It should return a 200 status, when all data are note updated', (done) => {
+    chai.request(myserver)
+      .patch('/api/v1/accommodations/1')
+      .set('Authorization', loggedtoken)
+      .field('address', 'kigali')
+      .field('name', 'hotel')
+      .end((err, res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body).to.have.property('msg');
+        expect(res.body).to.have.property('status');
```
Comment (elchroy) [inline src/tests/testAccomod.spec.js:77]: Assert that the fields are actually updated. `expect that address if 'kigali'`, etc. Also for name, and other test cases

File: src/tests/testAccomod.spec.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+      });
+  });
+  it('It should return a 200 status, when an accommodation has been deleted', (done) => {
+    chai.request(myserver)
+      .delete('/api/v1/accommodations/1')
+      .set('Authorization', loggedtoken)
+      .end((err, res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body).to.have.property('msg');
+        expect(res.body).to.have.property('status');
```
Comment (elchroy) [inline src/tests/testAccomod.spec.js:151]: Check that there is no accommodation with `id` of 1, after making this delete request

File: src/validation/index.js [hunk 5]
```diff
@@ -85,8 +85,23 @@ const validator = {
       const message = (Object.keys(req.error).length === 0) ? err.message : req.error;
       return sendResult(res, 400, message);
     }
-  }
+  },
 
+  editAccommodation(req, res, next) {
+    const valid = [
+      new Check({ name: req }).str().min(5),
```
Comment (elchroy) [inline src/validation/index.js:93]: Include the name field validation to the `add-Accommodation`.

**Pull request merged**